index.js

```diff
@@ -59,14 +60,19 @@ module.exports = fp(async function (app, opts) {
 
   const root = {}
   let schema = opts.schema
+  let gateway = opts.gateway
+
+  if (gateway && (schema || opts.resolvers || opts.loaders || opts.subscription)) {
+    throw new Error('Adding "schema", "resolvers", "loaders" or "subscription" to plugin options when plugin is running in gateway mode is not allowed')
+  }
```

Comment:
There are a few methods to change the schema/resolvers/loaders etc at aruntime, you'll need to throw in them as well.